apps/test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx

```diff
@@ -98,6 +99,27 @@ describe('useSchoolInfo', () => {
     expect(result.current.schoolZip).toBe('');
   });
 
+  it('should return a schoolName if no schoolId is passed', async () => {
+    const {result, waitForNextUpdate} = renderHook(() =>
+      useSchoolInfo({
+        ...initialState,
+        schoolId: '',
+      })
+    );
+    await waitForNextUpdate();
+
+    expect(result.current.schoolName).toBe(initialState.schoolName);
+  });
+
+  it('should not return a schoolName if schoolId is passed', async () => {
+    const {result, waitForNextUpdate} = renderHook(() =>
+      useSchoolInfo(initialState)
+    );
+    await waitForNextUpdate();
+
+    expect(result.current.schoolName).toBe('');
+  });
```

Comment:
Just to clarify my understanding of these tests: if no school id is passed in, then we know the school id wasn't updated so we can update the school name safely. However, if a school id is passed in then we know it's a different school so we want to have a clear school name?

Reply:
Oh wait on a reread I think I understand now. We return a `schoolName` when the user entered it, but not if it's pulled directy from the NCES data?

Reply:
yep! when we fetch the user's existing school info, we return either the nces school name, or the one the user entered manually. when we pass that existing data to the `useSchoolInfo` hook, we need to add the existing school name to the form only for manually entered names. for nces schools, the school name in the dropdown is actually derived from the school zip code search results, finding a match with the school id